objcgen/driver.cs

```diff
@@ -258,7 +277,7 @@ public void SetCompilationTarget (string value)
 		public List<Assembly> Assemblies { get; private set; } = new List<Assembly> ();
 		public Platform Platform { get; set; } = Platform.macOS;
 		public TargetLanguage TargetLanguage { get; private set; } = TargetLanguage.ObjectiveC;
-		public CompilationTarget CompilationTarget { get; set; } = CompilationTarget.SharedLibrary;
+		public CompilationTarget CompilationTarget { get; set; } = CompilationTarget.Framework;
```

Comment:
I think we still want the _normal_ `macOS` option to remains a `.dylib`. In fact it would be more correct to give a similar error if another option is given (for `macOS`) since it's, at least current, the only supported/available option.